third-party/rsutils/include/rsutils/subscription.h

```diff
@@ -0,0 +1,62 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "deferred.h"
+
+#include <functional>
+#include <memory>
+
+
+namespace rsutils {
+
+
+// Subscriptions can be temporary and so need to be removable. I.e., the subscriber needs to keep something (the
+// "subscription") around so that it may be removed later on. This can be captured via some value, or an object with
+// automatic destruction:
+//
+//     auto subscription = signal.subscribe( [] {} );
+// 
+// Note that cancellation is automatic on destruction, so it can be used in RAII.
+// 
+// Commonly used with signal.
+//
+class subscription
```

Comment:
Consider adding a "notify" method, enabling calling the deffered function not only on destruction.

Reply:
That's what `cancel()` is... right?

Reply:
`cancel()` is one time only because it detaches the function

Reply:
You want to call it multiple times? 
Then you can detach and store it elsewhere. The whole point of a subscription is that the unsubscribe action happens once and only once...